### PR TITLE
[ASCII-1219] Create a trace log component

### PR DESCRIPTION
**Pull request by pgimalac**
### What does this PR do?

Move the trace logger component from `comp/core/log` to its own package in `comp/core/log/tracelogimpl`.

### Motivation

Decouple trace-agent from core-agent. The end goal would be able to be able to build the serverless agent without the trace agent.

It also removes a lot of dependencies from some modules (which don't otherwise depend on `pkg/trace`), and in turn from some agents (see https://github.com/DataDog/datadog-agent/pull/23004#issuecomment-1954501638): 35~38 packages removed from cluster-agent, dogstatsd, process-agent, heroku agent, security-agent, and system-probe.

### Additional Notes

There shouldn't be any functional change.

### Describe how to test/QA your changes

Check that the trace-agent can still start and log properly.

### CODE REVIEW

File: comp/trace/log/trace_logger.go [hunk 1]
```diff
@@ -86,24 +87,24 @@ func (*tracelogger) Criticalf(format string, params ...interface{}) error {
...
 
-func newTraceLogger(lc fx.Lifecycle, params Params, config config.Component, telemetryCollector telemetry.TelemetryCollector) (log.Component, error) {
+func newTraceLogger(lc fx.Lifecycle, params corelogimpl.Params, config config.Component, telemetryCollector telemetry.TelemetryCollector) (log.Component, error) {
 	return NewTraceLogger(lc, params, config, telemetryCollector)
 }
 
 // NewTraceLogger creates a pkglog.Component using the provided config.LogConfig
-func NewTraceLogger(lc fx.Lifecycle, params Params, config config.LogConfig, telemetryCollector telemetry.TelemetryCollector) (log.Component, error) {
-	if params.logLevelFn == nil {
+func NewTraceLogger(lc fx.Lifecycle, params corelogimpl.Params, config config.LogConfig, telemetryCollector telemetry.TelemetryCollector) (log.Component, error) {
```
Comment (ogaca-dd) [inline comp/trace/log/trace_logger.go:95]: As `NewTraceLogger` is a constructor for `log.Componnent`, I think it makes sense  this code to be located in `core/log/impl` in a new package. What do you think?

Comment (pgimalac) [reply comp/trace/log/trace_logger.go:95]: I'll update that 👍

**Review by vickenty — APPROVED** [review]

**Review by mx-psi — APPROVED** [review]
Approving for OpenTelemetry team owned files

**Review by knusbaum — APPROVED** [review]
This change is fine with me. It looks like a strict improvement over what was there.

I'm wondering why we're moving it to another package under `comp/core` though, doesn't this belong under `comp/trace`?

**Review by ogaca-dd — APPROVED** [review]

**Pull request merged**